test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Utility/FileWatchers/UserConfigFileWatcherTests.cs

```diff
@@ -0,0 +1,192 @@
+            // File system watchers are not real time, so we need to give a little time for all the async file IO to happen
+            bool obtained = mre.Wait(TimeSpan.FromSeconds(10));
+
+            // Assert
+            obtained.Should().BeTrue();
+        }
+
+        [Fact]
+        public void FileChanged_WhenFileDeleted_NotificationReceivedAsync()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            string configPath = Path.Combine(testDirectory.Path, Settings.DefaultSettingsFileName);
+            File.WriteAllText(configPath, string.Empty);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            using UserConfigFileWatcher target = new(testDirectory.Path);
+            target.FileChanged += (s, e) => mre.Set();
+
+            // Act
+            mre.Wait(0).Should().BeFalse();
+            File.Delete(configPath);
+            // File system watchers are not real time, so we need to give a little time for all the async file IO to happen
+            bool obtained = mre.Wait(TimeSpan.FromSeconds(10));
+
+            // Assert
+            obtained.Should().BeTrue();
+        }
+
+        [Theory]
+        [InlineData("nuget.config", "nuget.config.old")]
+        [InlineData("nuget.config.old", "nuget.config")]
+        public void FileChanged_WhenFileRenamed_NotificationReceived(string filename1, string filename2)
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            string configPath1 = Path.Combine(testDirectory.Path, filename1);
+            string configPath2 = Path.Combine(testDirectory.Path, filename2);
+            File.WriteAllText(configPath1, string.Empty);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            using UserConfigFileWatcher target = new(testDirectory.Path);
+            target.FileChanged += (s, e) => mre.Set();
+
+            // Act
+            mre.Wait(0).Should().BeFalse();
+            File.Move(configPath1, configPath2);
+            // File system watchers are not real time, so we need to give a little time for all the async file IO to happen
+            bool obtained = mre.Wait(TimeSpan.FromSeconds(10));
+
+            // Assert
+            obtained.Should().BeTrue();
+        }
+
+        [Fact]
+        public void FileChanged_WhenFileCreatedInConfigSubdirectory_NotificationReceived()
+        {
+            // Arrange
+            using TestDirectory testDirectory = TestDirectory.Create();
+            var configDirectory = Path.Combine(testDirectory, "config");
+            using UserConfigFileWatcher target = new(testDirectory.Path);
+            ManualResetEventSlim mre = new(initialState: false, spinCount: 0);
+            target.FileChanged += (s, e) => mre.Set();
+            string configPath = Path.Combine(configDirectory, "contoso.config");
```

Comment:
I still don't understand why we need config directory in user config file watcher given by default it points to NuGetFolderPath.UserSettingsDirectory.

Reply:
In NuGet 5.7, user `config\*.config` support was added. Have a look at:
* https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#additional-user-wide-configuration
* https://github.com/NuGet/docs.microsoft.com-nuget/commit/fcf9f7d3b4f5768ca26b5f840e330b22b21e799b
* https://github.com/NuGet/NuGet.Client/pull/3421
* https://github.com/NuGet/Home/issues/9394